membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -58,10 +63,11 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext) exten
       val maybeAttributes = maybeZAttributes.map(ZuoraAttributes.asAttributes(_))
       attributesFromDynamo.foreach(maybeUpdateCache(maybeZAttributes, _, maybeAttributes))
       Future.successful("Zuora", maybeAttributes)
-    }).leftMap { zuoraError => // fallback to Dynamo if Zuora fails
-      SafeLogger.error(scrub"Failed to retrieve attributes from Zuora for $identityId because $zuoraError so falling back on cache.")
-      attributesFromDynamo.map(maybeDynamoAttributes => ("Dynamo", maybeDynamoAttributes.map(DynamoAttributes.asAttributes(_))))
-    }.merge.flatten // both branches return same type so can be merged
+    }) // fallback to cache if Zuora networking error or un-parsable response
+      .leftMap(fallbackToCache) // handles un-parsable response etc.
+      .merge // both branches return the same type so we can merge
```

Comment:
nice comment 😂

Reply:
PS I realise it was already there

Reply:
My thinking was `merge` might be unfamiliar so try to be explicit about it.